seL4-platforms/builds.py

```diff
@@ -178,14 +178,25 @@ def is_mcs(self) -> bool:
     def can_smp(self) -> bool:
         return self.get_mode() in self.get_platform().smp
 
-    def set_smp(self) -> bool:
+    def set_smp(self):
         if not self.can_smp():
             raise ValidationException("not Build.can_smp()")
         self.settings['SMP'] = "TRUE"
 
     def is_smp(self) -> bool:
         return self.settings.get('SMP') != None
 
+    def can_domains(self) -> bool:
+        return not self.is_smp()
+
+    def set_domains(self):
+        if not self.can_domains():
+            raise ValidationException("not Build.can_domains()")
+        self.settings['DOMAINS'] = "TRUE"
+
+    def is_domains(self) -> bool:
+        return self.settings.get('DOMAINS') != None
```

Comment:
my memories are that comparison to None should be done via `is None`, but they're hazy

Reply:
In this case they're going to be equivalent, but you're right that `.. is not None` would be better style here.

Reply:
then again the SMP test above has `!= None` so up to you if it's worth it

Reply:
I've added a commit that updates all of them.